src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -124,6 +124,9 @@ static void HandleNodeIdResolve(void * context, DnssdService * result, CHIP_ERRO
 #endif
     proxy->OnNodeIdResolved(nodeData);
     proxy->Release();
+
+    // Allocated in OnDnsResolveResult(), must be freed here
+    free(result);
```

Comment:
DnsResult is allocated using 
`Platform::New<DnsResult>`
Should we not use `Platform::Delete<DnsResult>(result);` here ?

Reply:
The DnsResult that was allocated with Platform::New is deleted with Platform::Delete in `GenericThreadStackManagerImpl_OpenThread<ImplClass>::DispatchResolve`.

But I don't understand this part at all.  `DnsResult` has a `DnssdService` member (by value, not pointer).  `GenericThreadStackManagerImpl_OpenThread<ImplClass>::::DispatchResolve` passes `&(dnsResult->mMdnsService)` to the callback.  That callback is this function.  So if we `free` the `DnssdService` here we are freeing memory that was never allocated, in the openthread case.

I just checked the Darwin impl, and there the `DnssdService` is just allocated on the stack before calling the callback.  So this `free` would be trying to free stack memory.

Reply:
I removed the free() call. This was a mistake.

Reply:
My mistake, I suggested this changed. I overlooked the `Platform::Delete` call in `DispatchResolve`.